### ISSUE TITLE
Type mismatch not detected when spreading Readonly generic compared to mutable version

### ISSUE THREAD

**Issue by b-gutzmann** [microsoft/TypeScript#62756]
### 🔎 Search Terms

Readonly<T> Readonly generic spread

### 🕗 Version & Regression Information

Typescript v5.9.3 
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about Spread, Generic, Readonly

### 💻 Code

```ts
interface Foo {
    bars: { bar: string }[];
}

interface Props {
    foo: Foo;
}

// Generic
function error<T extends Props>(myProps: T) {
    // Correctly has an error
    const fooBad: Foo = { ...myProps.foo, bars: 'I should not be a string' };
    //                                    ^^^^
    // Type 'string' is not assignable to type '{ bar: string; }[]'.ts(2322)
}

// Readonly
function error2(myProps: Readonly<Props>) {
    // Correctly has an error
    const fooBad: Foo = { ...myProps.foo, bars: 'I should not be a string' };
    //                                    ^^^^
    // Type 'string' is not assignable to type '{ bar: string; }[]'.ts(2322)
}

// Generic + Readonly
function noError<T extends Props>(myProps: Readonly<T>) {
    // No error here!
    const fooBad: Foo = { ...myProps.foo, bars: 'I should not be a string' };
}
```

### 🙁 Actual behavior

Typescript only reports type errors with  `myProps: T` and `myProps: Readonly<Props>` and *not* `myProps: Readonly<T>`

### 🙂 Expected behavior

Expect a type error to be reported in all cases `myProps: T`, `myProps: Readonly<Props>`, and `myProps: Readonly<T>`

**Comment by LukeAbby** [comment]
This is actually a more general problem.
```ts
type MappedType<T> = { [K in keyof T]: T[K] }

function noError<T extends Props>(myProps: MappedType<T>) {
    // No error here!
    const fooBad: Foo = { ...myProps.foo, bars: 'I should not be a string' };
}
```

It's actually about spreading a generic property of any mapped type, not just `Readonly`.

@Andarist pointed out that the type of `bars` ends up as `T["foo"] & string` after spreading. Since `T["foo"]` is `{ bars: string []}` this makes it legal to assign to `{ bars: { bar: string }[] }`

**Comment by MartinJohns** [comment]
Duplicate of #42690.
